File: 338.counting-bits.py
#
# @lc app=leetcode id=338 lang=python
#
# [338] Counting Bits
#

# @lc code=start
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        returnList = []
        for i in range(0,n+1):
            returnList.append(bin(i).count('1'))
        return returnList
# @lc code=end

File: 338.counting-bits.java
/*
 * @lc app=leetcode id=338 lang=java
 *
 * [338] Counting Bits
 */

// @lc code=start
class Solution {
    public int[] countBits(int n) {
        int[] returnArray = new int[n+1];
        for (int i = 0; i<=n;i++) {
            int count = 0;
            for (char a : Integer.toBinaryString(i).toCharArray()) {
                if (a == '1') {
                    count++;
                }
            }
            returnArray[i] = count;
        }
        return returnArray;
    }
}
// @lc code=end

